### DISCUSSION TITLE
context values

### DISCUSSION THREAD

**Discussion by andrconstruction** [react-hook-form/react-hook-form#1988] [Q&A] [answered]
Hello All, I have next code

 <Form.Group>
                <Form.Label>Логин</Form.Label>
                <p>{user.id}</p>
                <p>{user.username}</p>

                <Controller
                    as={<Form.Control isValid={dirty.username && !errors.username} isInvalid={errors.username}
                                      placeholder={"login"}/>}
                    rules={{
                        required : "обязательное поле"
                        , maxLength : 255, minLength : 2
                    }}
                    name="username"
                    control={control}
                    defaultValue={user.username}
                />
                <ErrorMessage
                    errors={errors}
                    as={<small/>}
                    style={{color : "red"}}
                    name="username"
                />
            </Form.Group>

User data comming from a context, and  under the label component show as expected :
                <p>{user.id}</p>
                <p>{user.username}</p>

First time when form rendered defaultValue is null and when I select next user defaultValue has a previous user data it is a bug or I make some wrong?

**Comment by bluebill1049** [comment, marked as answer]
please provide a codesandbox.

**Reply by andrconstruction** [reply]
Too many code should to extract from project. As I suspect problem with async values passed to a form. How to do it right?

**Reply by andrconstruction** [reply]
reset save my life
